quinn-proto/src/config.rs

```diff
@@ -434,6 +434,11 @@ where
     /// Improves behavior for clients that move between different internet connections or suffer NAT
     /// rebinding. Enabled by default.
     pub(crate) migration: bool,
+
+    /// Whether to use the local IP address which was used when receiving the initial
+    /// packet to send out all outgoing packets to a clients instead of the IP
+    /// the the socket is bound to.
+    pub(crate) use_intial_local_ip_on_send: bool,
```

Comment:
Are there any compelling reasons not to do this on platforms where it is available? Should we default to true? Also, how about a slightly less verbose name, maybe `send_from_initial_ip`?

(Also you have a consistent typo, `intial` -> `initial`.)

Reply:
`send_from_initial_ip` is fine to me. I don't see a huge amount of reasons why it shouldn't be the default. But I tried to avoid breaking people by enabling it unconditionally. We can also set the default to `true`, and allow people to disable it?

Reply:
Do you see any reason? I'd lean towards not making it configurable by default, until someone complains. Easier to add the API later than remove it. (But curious what @Ralith thinks.)

Reply:
The only reason I see is "I haven't tested on enough platforms". I know it works fine on my dev machine as well as a test server I deploy too, but can't comment on anything else. I'm good with removing too, and adding more APIs if deemed necessary.

I guess one reason could be "if bound to only one socket, attaching the cmsg to every outgoing packet might be more work". But I doubt that is quantifyable.

Reply:
AFAICT this change is purely a bugfix and disabling it would just reintroduce the bug, so I don't see a reason for it to be optional.

Reply:
Ok. Should I remove the switch then?

Reply:
Yup, let's try it!